Playwright: migrate spec for Signup: Domains Only flow.

worldomonation opened the pull request:
#### Changes proposed in this Pull Request

This PR migrates the Signup: Domains Only flow from the Selenium suite.

Key changes:
- new methods in `DataHelper` to generate test domain registrar details.
- new methods in relevant pages to enter forms.

Details:

This migration stayed faithful to the Selenium version of the test. The only additional step tested is the `Turn of auto-renew` step to verify interaction with the 'purchased' domain.

During the development of this test case, I ran across an issue with the post-purchase page:
https://github.com/Automattic/wp-calypso/issues/56548


#### Testing instructions

- [x] pr tests
  - [x] mobile
  - [x] desktop
- [x] pre-release tests 

Related to #

worldomonation committed:
Implement spec for Signup: Domains only.

worldomonation committed:
Include step to submit the registrar details

worldomonation committed:
Correct the flow after checkout:
- make it wait for the redirect

worldomonation commented:
Selenium failure are unrelated.

WunderBart commented on test/e2e/specs/specs-playwright/wp-signup__domain.ts:
```diff
@@ -0,0 +1,134 @@
+
+		it( 'Prices are shown in Japanese Yen', async function () {
+			const buttonText = await cartCheckoutPage.getPaymentButtonText();
+			buttonText.startsWith( '¥' );
```
Looks like this should be wrapped with `expect`

worldomonation replied:
Haha, oops...

worldomonation replied:
Found out this snippet was not actually doing anything useful, and it would have failed if wrapped with `expect` because the actual button text reads something like `Pay A$120`, so the assertion will always fail.

I've had to rewrite this step to read the value from the checkout total, but the resulting step runs and passes as expected.

WunderBart reviewed (approved):
Nice work! Just one tiny comment 🚀

worldomonation committed:
Fix faulty logic around checking currency symbol.
The checkout button starts not with the currency symbol, but with the text `Pay`, so the step that checks currency symbol will always fail.
